types/chrome/index.d.ts

```diff
@@ -2,6 +2,9 @@
 /// <reference path="./har-format/index.d.ts" />
 /// <reference path="./chrome-cast/index.d.ts" />
 
+// NoInfer for old TypeScript versions
+type NoInferX<T> = T[][T extends any ? 0 : never];
```

Comment:
This type seems to have been common before `NoInfer` was officially added to  TypeScript in version 5.4. I found it here: https://dev.to/allieschen/behind-noinfer-in-typescript-319

Please let me know if there's a better way to do this!

Actually, most people used this:

```typescript
type NoInferX<T> = [T][T extends any ? 0 : never];
```

But the linter complained about `[T]` so I had to make that `T[]`. The error was:

```
  6:20  error  Type [T] is a single-element tuple type. You probably meant T[]  @definitelytyped/no-single-element-tuple-type
```